txpool/txpool.go

```diff
@@ -473,9 +485,11 @@ func (t *TxPool) ProcessEvent(evnt *blockchain.Event) {
 
 	// remove the mined transactions from the pendingQueue list
 	for _, txn := range delTxns {
-		t.decreaseSlots(numSlots(txn))
-		t.pendingQueue.Delete(txn)
-		t.remoteTxns.Delete(txn)
+		deletedTx := t.pendingQueue.Delete(txn)
+		if deletedTx != nil {
+			t.gauge.decrease(slotsRequired(txn))
+			t.remoteTxns.Delete(txn)
+		}
```

Comment:
```suggestion
		if deletedTx := t.pendingQueue.Delete(txn); deletedTx != nil {		        
			t.gauge.decrease(slotsRequired(txn))
			t.remoteTxns.Delete(txn)
		}
```

Just a minor nitpick 🙂